beetsplug/unimported.py

```diff
@@ -40,11 +40,19 @@ def commands(self):
         def print_unimported(lib, opts, args):
             ignore_exts = [('.' + x).encode() for x
                            in self.config['ignore_extensions'].as_str_seq()]
-            in_folder = {
-                os.path.join(r, file) for r, d, f in os.walk(lib.directory)
-                for file in f if not any(
-                    [file.endswith(extension) for extension in
-                     ignore_exts])}
+            ignore_dirs = [lib.directory + ('/' + x.strip('/')).encode() for x
+                           in self.config['ignore_subdirectories'].as_str_seq()]
+
+            in_folder = set()
+            for r, d, f in os.walk(lib.directory):
+                if r in ignore_dirs:
+                    continue
+                files = {
+                    os.path.join(r, file)
+                    for file in f if not 
+                    any([file.endswith(ext) for ext in ignore_exts])
+                }
+                in_folder = in_folder | files
```

Comment:
Maybe I'm missing something obvious, but was there a specific reason to switch away from the set comprehension and to use a `for` loop instead? It seems like adding `r not in ignore_dirs and...` would have the desired effect, right?

Reply:
Honestly, just for readability. I'm happy to keep comprehensions for the sake of speed, but I feel like it makes the code less readable. What do you think?

Reply:
Ah, oddly enough, I actually found the comprehension version easier to read (because it's closer to "declaratively" writing down what the desired set contains). It could certainly use some additional whitespace/line-breaks, but can we give it a shot and see how bad it looks?